### PR TITLE
Improve debug window

### LINKED ISSUE #4167: Debug window won't scroll
The debug window aggressively scrolls to the top of the window.

To reproduce: Use a debug build, click on a field, choose the Watch tab, click the DBG button, try to scroll down in the field-debug window.

### PR DESCRIPTION
**Pull request by matthiakl**
* Fix scrolling (#4167): update text only when changed
* Open bobs with double click: with selection every bob is opened when scrolling through using the mouse wheel
* Change structs to classes (relates to #1558)

### CODE REVIEW

File: src/wui/game_debug_ui.cc [hunk 1]
```diff
@@ -339,7 +281,10 @@ void FieldDebugWindow::think() {
 		}
 	}
 
-	ui_field_.set_text(str.c_str());
+	if (str.compare(text_) != 0) {
```
Comment (gunchleoc) [inline src/wui/game_debug_ui.cc]:
```suggestion
	if (!text_.empty()) {
```

**Review by gunchleoc — COMMENTED** [review]
1 code style nit, rest LGTM :)

File: src/wui/game_debug_ui.cc (hunk 1 above)
Comment (matthiakl) [reply src/wui/game_debug_ui.cc]: Then the text would only be set during the first call, but will not update if the field properties change, while the window is open.

Comment (gunchleoc) [reply src/wui/game_debug_ui.cc]: I can never remember what 0 means in the string compare... `if (text_ == str)`? Strings have a direct equality operator these days, so there's no need for using the compare function.

Comment (matthiakl) [reply src/wui/game_debug_ui.cc]: 0 means equal, therefor `(str.compare(text_) != 0)` is the same as `(str != text_)`

**Review by gunchleoc — APPROVED** [review]

**Pull request merged**